### PR TITLE
EnzoInitialShuCollapse parameter refactorization

### PR DESCRIPTION
**Pull request by jakereinheimer**
### Pull request summary

Refactoring EnzoInitialShuCollapse parameters. Addresses issue https://github.com/enzo-project/enzo-e/issues/383. Short because I am going through it live with Matthew

### CODE REVIEW

**Comment by jwise77** [comment]
There are some compilation errors
```
/home/circleci/enzo-e/src/Enzo/initial/EnzoInitialShuCollapse.cpp:21:4: error: expected ‘{’ before ‘center_’
   21 |    center_{}
      |    ^~~~~~~
/home/circleci/enzo-e/src/Enzo/initial/EnzoInitialShuCollapse.cpp: At global scope:
/home/circleci/enzo-e/src/Enzo/initial/EnzoInitialShuCollapse.cpp:21:4: error: ‘center_’ does not name a type
/home/circleci/enzo-e/src/Enzo/initial/EnzoInitialShuCollapse.cpp:22:4: error: ‘drift_velocity_’ does not name a type
   22 |    drift_velocity_{}
      |    ^~~~~~~~~~~~~~~
/home/circleci/enzo-e/src/Enzo/initial/EnzoInitialShuCollapse.cpp:23:1: error: expected unqualified-id before ‘{’ token
   23 | {
      | ^
```

File: src/Enzo/initial/EnzoInitialShuCollapse.cpp [hunk 1]
```diff
@@ -10,31 +10,24 @@
...
-   instability_parameter_(instability_parameter),
-   external_density_(external_density),
-   central_sink_exists_(central_sink_exists),
-   central_sink_mass_(central_sink_mass)
+   truncation_radius_(p.value_float("truncation_radius",0.0)),
+   nominal_sound_speed_(p.value_float("nominal_sound_speed",0.0)),
+   instability_parameter_(p.value_float("instability_parameter",0.0)),
+   external_density_(p.value_float("external_density",0.0)),
+   central_sink_exists_(p.value_logical("central_sink_exists",false)),
+   central_sink_mass_(p.value_float("central_sink_mass",0.0))
```
Comment (mabruzzo) [inline src/Enzo/initial/EnzoInitialShuCollapse.cpp:20]:
```suggestion
   central_sink_mass_(p.value_float("central_sink_mass",0.0)),
```

File: src/Enzo/initial/EnzoInitialShuCollapse.cpp [hunk 2]
```diff
@@ -10,31 +10,24 @@
...
+   center_{}
```
Comment (mabruzzo) [inline src/Enzo/initial/EnzoInitialShuCollapse.cpp:21]:
```suggestion
   center_{},
```

**Review by mabruzzo — CHANGES_REQUESTED** [review]
This is fantastic! I really appreciate you doing this!

I highlighted 2 missing commas that are needed in order to get this to compile properly.

There are also 2 other changes that are necessary:
1. The order in which the members appear in the "member-initializer-list" in the constructors should match the order in which they are declared within the class itself (jump to the end of `EnzoInitialShuCollapse` to see this ordering).
    - technically, this is not strictly necessary, but it's good practice (if you don't do this, you can get unexpected behavior while using one member to initialize another member) 
    - plus `g++` raises some annoying warnings if you don't do this
2. More importantly, you are changing the default value of the parameters
    - this is my fault for not explaining this.
    - When you call something like `p.value_float("truncation_radius", ...)` the trailing value represents the default value.
    - Currently, you've changed the default values compared to what was previously used within `EnzoConfig::read_initial_shu_collapse_`

**Pull request closed without merging**